frontend/packages/console-plugin-sdk/src/codegen/plugin-integration-tests.ts

```diff
@@ -0,0 +1,39 @@
+import * as _ from 'lodash';
+import {
+  PluginPackage,
+  filterActivePluginPackages,
+  PluginPackageFilter,
+  resolvePluginPackages,
+} from './plugin-resolver';
+
+const CONSOLE_PLUGIN_PREFIX = '@console/';
+const CONSOLE_INEGRATION_TESTS_PREFIX = '@console/integration-tests/';
+
+/**
+ * Return a map of integration tests.
+ */
+export const pluginIntegrationTests = (
+  monorepoRootDir: string = process.cwd(),
+  pluginFilter: PluginPackageFilter = filterActivePluginPackages,
+) => {
+  const shortName = (plugin: PluginPackage) => plugin.name.slice(CONSOLE_PLUGIN_PREFIX.length);
+
+  const integrationTests = (plugin: PluginPackage) =>
+    plugin.name.startsWith(CONSOLE_PLUGIN_PREFIX) &&
+    plugin.consolePlugin.integrationTests &&
+    _.mapValues(plugin.consolePlugin.integrationTests, (paths) =>
+      _.map(paths, (path) =>
+        path.startsWith(CONSOLE_INEGRATION_TESTS_PREFIX)
```

Comment:
This code resolves package paths to actual paths _which are relative to root `integration-tests` directory_, so I'd move this function into `integration-tests/protractor.conf.ts`.

Also, the `shortName` function isn't usable if plugin packages won't use the `@console` scope.

I think the best solution is to enhance our `Package` type to include path of the file it was loaded from.

So in `packages/console-plugin-sdk/src/codegen/plugin-resolver.ts`:

```ts
export type Package = readPkg.NormalizedPackageJson & {
  _path: string;
};
```

and update `readPackages` function (and its test) accordingly.

With that, the `shortName` function isn't really needed, and `monorepoRootDir` wouldn't be needed as well.

Reply:
done :tada: :cake: